lib/utils.py

```diff
@@ -24,9 +24,30 @@
 import subprocess
 import uuid
 import shutil
+import time
 from typing import Union, Dict, Any
 
 
+def cleandir(d: str) -> None:
+    with os.scandir(d) as it:
+        for entry in it:
+            os.unlink(f"{d}/{entry.name}")
+    os.rmdir(d)
+
+
+def cleanup(varg: Dict[str, Any]) -> None:
+    """cleanup submit directory etc."""
+    os.chdir(f'{varg["submitdir"]}/..')
+    cleandir(varg["submitdir"])
+    # now clean up old submit directories that weren't
+    # cleaned up by jobsub_submit at the time
+    with os.scandir(".") as it:
+        for entry in it:
+            sb = os.stat(entry.name)
+            if entry.name.startswith("js_") and sb.st_mtime - time.time() > 604800:
```

Comment:
Shouldn't this subtraction be reversed?  That is `time.time() - sb.st_mtime`?

Reply:
Ack!  yes

Reply:
Fixed.  I think I'm getting back all the files I should now for both dag and plain submits, so I think this is ready to merge.